src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,150 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+/**
+ * TaskList represents all the tasks contained in a Duke Chatbot.
+ *
+ * @author Jovyn Tan
+ * @version CS2103 AY21/22 Sem 1
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * A constructor for a duke.TaskList which contains Tasks.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Converts a taskList to a text format that can be saved in a txt file.
+     * @return a String that represents the savable data of a duke.TaskList.
+     */
+    public String toSaveData() {
+        String data = "";
+        for (Task task : this.tasks) {
+            data += task.toSaveData();
+        }
+        return data;
+    }
+
+    /**
+     * Adds an existing task to the list of tasks.
+     * @param task The task to be added to the list of tasks.
+     */
+    public void addTask(Task task) {
+        this.tasks.add(task);
+    }
+
+    /**
+     * Given a string, creates a To-do from that string and adds it to the list of task.
+     * @param taskTitle a String of the title of the To-do to be added.
+     * @return the newly created To-do
+     */
+    public String addNewTodo(String taskTitle) {
+        Todo task = new Todo(taskTitle);
+        this.tasks.add(task);
+        return "Got it. I've added this task:\n\t" + task.toString() + this.countTasks();
+    }
+
+    /**
+     * Given a string, creates a Deadline from that string and adds it to the list of task.
+     * @param taskTitle a String of the title of the Deadline to be added.
+     * @return the newly created Deadline.
+     */
+    public String addNewDeadline(String taskTitle) {
+        int delimiter = taskTitle.indexOf("/by ");
+        LocalDate due = LocalDate.parse(taskTitle.substring(delimiter + 4));
```

Comment:
Perhaps you could define a final constant that takes the value of 4 so we know what the value is referring to?